core/src/nl/tudelft/watchdog/core/logic/ui/events/WatchDogEventType.java

```diff
@@ -0,0 +1,268 @@
+package nl.tudelft.watchdog.core.logic.ui.events;
+
+import nl.tudelft.watchdog.core.logic.document.Document;
+import nl.tudelft.watchdog.core.logic.document.EditorWrapperBase;
+import nl.tudelft.watchdog.core.logic.interval.IDEIntervalManagerBase;
+import nl.tudelft.watchdog.core.logic.interval.intervaltypes.*;
+import nl.tudelft.watchdog.core.logic.ui.InActivityNotifiers;
+
+import java.util.Date;
+
+/** The different type of events. */
+@SuppressWarnings("javadoc")
+public enum WatchDogEventType implements WatchDogEventTypeInterface {
+
+    ACTIVE_WINDOW {
+        @Override
+        public void process(Date forcedDate, Object source) {
+            IDEActiveInterval interval = intervalManager.getInterval(IDEActiveInterval.class);
+            if (isClosed(interval)) {
+                intervalManager.addInterval(new IDEActiveInterval(forcedDate));
+            }
+            InActivityNotifiers.USER_INACTIVITY.trigger(forcedDate);
+        }
+    }, INACTIVE_WINDOW {
+        @Override
+        public void process(Date forcedDate, Object source) {
+            IDEActiveInterval interval = intervalManager.getInterval(IDEActiveInterval.class);
+            intervalManager.closeInterval(interval, forcedDate);
+        }
+    }, START_IDE {
+        @Override
+        public void process(Date forcedDate, Object source) {
+            intervalManager.addInterval(new IDEOpenInterval(forcedDate));
+            InActivityNotifiers.USER_INACTIVITY.trigger(forcedDate);
+        }
+    }, END_IDE {
+        @Override
+        public void process(Date forcedDate, Object source) {
+            InActivityNotifiers.USER_INACTIVITY.cancelTimer(forcedDate);
+        }
+    },
+
```

Comment:
Can you separate the enum types like this for readability throughout the document? Thanks!

Reply:
Maybe adding comments to understand the separation of the different subtypes helps

Reply:
I have separated the enum types with one new line between each type